### PR TITLE
Add locations permissions setting

### PR DESCRIPTION
**Pull request by NickAb**
Add ability to set location permissions.

We use wix/AppleSimulatorUtils (0.6.4 or higher) to set location permissions.

Probably we should update service permissions setter to use wix/AppleSimulatorUtils too, so we can delete and don't maintain sqlite TCC related code.

I removed the initially added ability to set notifications permissions as it requires SpringBoard restart which takes about 10 seconds and also breaks WebDriverAgent. So we will have to restart it too, adding even more time, making handling of alert manually faster and easier.

### CODE REVIEW

File: device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/simulator/SimulatorPermissions.kt [hunk 1]
```diff
@@ -56,4 +64,33 @@ class SimulatorPermissions(
             throw(SimulatorError("Failed to update type $type for $this"))
         }
     }
+
+    private val appleSimUtils = "/usr/local/bin/applesimutils"
+
+    @Suppress("UNUSED_PARAMETER")
+    private fun setNotificationsPermission(bundleId: String, value: PermissionAllowed) {
+        // Setting notifications permission is disallowed as it results in SpringBoard restart
+        // which breaks WebDriverAgent. Restarting SpringBoard and WebDriverAgent will take to much time.
```
Comment (fr0l) [inline device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/simulator/SimulatorPermissions.kt]: to -> too

File: device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/simulator/SimulatorPermissions.kt [hunk 2]
```diff
@@ -56,4 +64,33 @@ class SimulatorPermissions(
...
+        throw RuntimeException("Setting notifications permission is not supported")
+    }
+
+    private fun setLocationPermission(bundleId: String, value: PermissionAllowed) {
```
Comment (fr0l) [inline device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/simulator/SimulatorPermissions.kt]: 1. rename 'value' to permissionAllowed.
2. use strings from enum

```
        val permissionValue = when (permissionAllowed) {
            Always, Inuse, Never, Unset -> permissionAllowed.value
            else -> throw IllegalArgumentException("")
        }
```

File: Brewfile [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[
+    "homebrew/Brewfile",
+    "homebrew/Brewfile.devices",
+].each do |f|
+    eval(File.new(f).read)
+end
```
Comment (fr0l) [inline Brewfile:6]: doesn't look like related change. let's move to a separate pr

File: homebrew/Brewfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+tap "homebrew/core"
+
+tap "wix/brew"
+tap "facebook/fb"
+
+# Play, record, convert, and stream audio and video
+brew "ffmpeg"
+
+# Apple simulator utilities
+brew "wix/brew/applesimutils"
```
Comment (fr0l) [inline homebrew/Brewfile:11]: doesn't look like related change. let's move to a separate pr

File: homebrew/Brewfile.devices [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+# but if they are already installed, brew will not auto-fetch new head
+
+# USB multiplexor daemon for iPhone and iPod Touch devices
+brew "usbmuxd", args: ["HEAD"]
+
+# Library to communicate with iOS devices natively
+brew "libimobiledevice", args: ["HEAD"]
+
+# netcat on steroids
+brew "socat"
```
Comment (fr0l) [inline homebrew/Brewfile.devices:14]: doesn't look like related change. let's move to a separate pr

**Review by fr0l — CHANGES_REQUESTED** [review]

**Comment by fr0l** [comment]
> Probably we should update service permissions setter to use wix/AppleSimulatorUtils too, so we can delete and don't maintain sqlite TCC related code.

👍

File: device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/simulator/SimulatorPermissions.kt (hunk 2 above)
Comment (NickAb) [reply device-server/src/main/kotlin/com/badoo/automation/deviceserver/ios/simulator/SimulatorPermissions.kt]: I don't think we should use strings from enum here. Enum string values are DTOs and used on the client side, while strings we use here are internal implementation specific to applesimutils. And its just a coincidence that applesimutils use the same values as our DTO.

File: homebrew/Brewfile (hunk 4 above)
Comment (NickAb) [reply homebrew/Brewfile:11]: applesimutils are required for this PR to work, so I think a way to describe such dependency fits as part of PR. I've separated this into a separate commit.
